### PR TITLE
✨ Added support for variable substitution of IANA's Timezone Code

### PR DESCRIPTION
**Pull request by Manugal**
Some publishers would like to collect timezone code, because it provides more precise information about an event's time.

Issue: #13044

### CODE REVIEW

**Comment by Manugal** [comment]
@zhouyx could you review this?

**Comment by Manugal** [comment]
@zhouyx ping :blush:

File: spec/amp-var-substitutions.md [hunk 1]
```diff
@@ -193,6 +193,7 @@ The tables below list the available URL variables grouped by type of usage. Furt
 | [Scroll Left](#scroll-left)         | `SCROLL_LEFT`     | `${scrollLeft}`     |
 | [Scroll Top](#scroll-top)           | `SCROLL_TOP`      | `${scrollTop}`      |
 | [Timezone](#timezone)               | `TIMEZONE`        | `${timezone}`       |
+| [Timezone Code](#timezoneCode)      | `TIMEZONE_CODE`   | `${timezoneCode}`   |
```
Comment (zhouyx) [inline spec/amp-var-substitutions.md]: nit: #timezone-code

File: src/service/url-replacements-impl.js [hunk 2]
```diff
@@ -352,6 +352,21 @@ export class GlobalVariableSource extends VariableSource {
     // Returns the user's time-zone offset from UTC, in minutes.
     this.set('TIMEZONE', dateMethod('getTimezoneOffset'));
 
+    // Returns the IANA timezone code
+    this.set('TIMEZONE_CODE', () => {
+      let tzCode = '';
+      if (Intl && typeof Intl === 'object') {
```
Comment (zhouyx) [inline src/service/url-replacements-impl.js]: Please check the existence of `Intl` to the current window object. 
Try 
```
if ('Intl' in this.ampdoc.win && 'DataTimeFormat' in this.ampdoc.win.Intl ...) {
  ...
}
```

File: src/service/url-replacements-impl.js [hunk 3]
```diff
@@ -352,6 +352,21 @@ export class GlobalVariableSource extends VariableSource {
...
+        tzCode = new Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+        // It could be undefined (i.e. IE11)
```
Comment (zhouyx) [inline src/service/url-replacements-impl.js]: Good catch! 
One question I have: when getting `timeZone` undefined in IE. Will it throw an error? Or it returns undefined silently? 

BTW: You can save this by returning `tzCode ? tzCode : ''`

File: test/functional/test-url-replacements.js [hunk 4]
```diff
@@ -479,6 +479,12 @@ describes.sandboxed('UrlReplacements', {}, () => {
     });
   });
 
+  it('should replace TIMEZONE_CODE', () => {
+    return expandUrlAsync('?tz_code=TIMEZONE_CODE').then(res => {
+      expect(res).to.match(/tz_code=\w+/);
```
Comment (zhouyx) [inline test/functional/test-url-replacements.js]: Since `TIMEZONE_CODE` can resolved to empty string. We may need to reflect that in the test.

**Comment by Manugal** [comment]
I need to close it and reopen a new one. Sorry.

**Pull request closed without merging**